server/channels/app/notification_push.go

```diff
@@ -138,6 +140,25 @@ func (a *App) sendPushNotificationToAllSessions(msg *model.PushNotification, use
 }
 
 func (a *App) sendPushNotification(notification *PostNotification, user *model.User, explicitMention, channelWideMention bool, replyToThreadType string) {
+	cancelled := false
+	a.ch.RunMultiHook(func(hooks plugin.Hooks) bool {
+		rejectionReason := hooks.NotificationWillBePushed(&model.PluginPushNotification{
+			Post:    notification.Post.ForPlugin(),
+			Channel: notification.Channel,
+			UserID:  user.Id,
+		})
+		if rejectionReason != "" {
+			mlog.Error("Notification rejected by plugin.", mlog.String("reason", rejectionReason))
```

Comment:
Instead of only rejecting, can we allow the plugin to modify the notification if necessary, like we do for `MessageWillBePosted`?

Reply:
I discussed this a bit in the summary (see point 3), but in short: I found it very awkward to allow a plugin to modify the notification, because there really isn't a notification at this point yet. So do we allow a plugin to change the post, the channel, etc? That's why I felt it was just easier to allow the plugin to create its own notification if it wanted something custom (like Calls does) instead of weirdly changing the base model structs to generate a different notification.